### PR TITLE
feat: APP-202 buy credits step 2

### PR DESCRIPTION
**Pull request by blushi**
## Description

https://regennetwork.atlassian.net/browse/APP-202?atlOrigin=eyJpIjoiMGZmODlhMTc4YTY5NGYxNWJmZWM3MWE0MDYzNTNmNDAiLCJwIjoiaiJ9

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

https://deploy-preview-2418--regen-storybook.netlify.app/?path=/story/marketplace-organisms-paymentinfoform--fiat-logged-out
https://deploy-preview-2418--regen-storybook.netlify.app/?path=/story/marketplace-organisms-paymentinfoform--fiat-logged-in-no-email
https://deploy-preview-2418--regen-storybook.netlify.app/?path=/story/marketplace-organisms-paymentinfoform--fiat-logged-in-with-email
https://deploy-preview-2418--regen-storybook.netlify.app/?path=/story/marketplace-organisms-paymentinfoform--fiat-logged-in-with-payment-method
https://deploy-preview-2418--regen-storybook.netlify.app/?path=/story/marketplace-organisms-paymentinfoform--crypto-no-email
https://deploy-preview-2418--regen-storybook.netlify.app/?path=/story/marketplace-organisms-paymentinfoform--crypto-with-email

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-storybook/.storybook/main.js [hunk 1]
```diff
@@ -69,11 +69,11 @@ module.exports = {
...
-        react({
-          babel: {
-            plugins: ['macros'],
-          },
-        }),
+        // react({
+        //   babel: {
+        //     plugins: ['macros'],
+        //   },
+        // }),
```
Comment (blushi) [inline web-storybook/.storybook/main.js:76]: commenting this for now, it's causing some issues running storybook locally
will need to investigate further cc/ @flagrede

File: web-marketplace/src/components/organisms/PaymentInfoForm/PaymentInfoForm.CustomerInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
+import { useFormContext, useWatch } from 'react-hook-form';
+
+// import { msg, Trans } from '@lingui/macro';
+// import { useLingui } from '@lingui/react';
```
Comment (blushi) [inline web-marketplace/src/components/organisms/PaymentInfoForm/PaymentInfoForm.CustomerInfo.tsx:4]: TODO use lingui once previous issue fixed

**Comment by blushi** [comment]
@erikalogie @clevinson see testing instructions

**Comment by erikalogie** [comment]
This looks great to me! 

One small design detail on mobile: 
Marketplace : Organisms : PaymentInfoForm - Fiat Logged Out ⋅ Storybook 2024-07-30 09-30-42

Also let's change the text under "Your name" to "This name will appear on the retirement certificate unless you choose to retire anonymously in the next step. It is also your user profile name." I think that is more clear.

**Comment by blushi** [comment]
@erikalogie done

**Comment by erikalogie** [comment]
LGTM!

File: web-marketplace/src/components/organisms/PaymentInfoForm/PaymentInfoForm.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      email: accountEmail,
+      name: accountName,
+      createAccount: true,
+      savePaymentMethod: true,
+      paymentMethodId: paymentMethods?.[0]?.id,
+    },
+    mode: 'onBlur',
+  });
+
+  const stripePromise =
```
Comment (r41ph) [inline web-marketplace/src/components/organisms/PaymentInfoForm/PaymentInfoForm.tsx:55]: Maybe we could use useMemo here?

File: web-marketplace/src/components/organisms/PaymentInfoForm/PaymentInfoForm.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    paymentOption === 'card' &&
+    stripePublishableKey &&
+    loadStripe(stripePublishableKey);
+
+  const options: StripeElementsOptionsMode = {
```
Comment (r41ph) [inline web-marketplace/src/components/organisms/PaymentInfoForm/PaymentInfoForm.tsx:60]: Maybe we could use useMemo here too?

File: web-marketplace/src/components/organisms/PaymentInfoForm/PaymentInfoForm.CustomerInfo.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+          description={`This name will appear on the retirement certificate unless you choose to retire anonymously in the next step. It is also your user profile name.`}
+          {...register('name')}
+          error={!!errors['name']}
+          helperText={errors['name']?.message}
+          disabled={!!accountName}
+        />
+      )}
+      <TextField
+        className={!accountId && !wallet ? 'mb-30' : ''}
+        label={`Your email`}
```
[the remark this answers is not in the document]
Comment (blushi) [reply web-marketplace/src/components/organisms/PaymentInfoForm/PaymentInfoForm.CustomerInfo.tsx:67]: I think I had started to include lingui in there then realized there was an issue with storybook, so just removed the wrapping `_(msg``)` from the prop value resulting in ` `` `
I can turn this into the simpler string form in the meantime, but we'll need to migrate to use lingui eventually

**Review by r41ph — APPROVED** [review]

**Pull request merged**